include/core/solid_objects_parameters.h

```diff
@@ -373,30 +400,56 @@ namespace Parameters
     Verbosity verbosity;
 
     // DEM solid objects
-    std::vector<std::shared_ptr<RigidSolidObject<dim>>> solids;
-    unsigned int                                        number_solids;
+    std::vector<std::shared_ptr<RigidSolidObject<dim>>> solids_2d;
+    std::vector<std::shared_ptr<RigidSolidObject<dim>>> solids_3d;
+    unsigned int                                        number_solids_2d;
+    unsigned int                                        number_solids_3d;
```

Comment:
Wait what, this is weird, why not template RigidSOlidObject by <dim,spacedim> and then the 2d would be <2,3> and the 3d would be <3,3> ?
Furthermore, I would refer to dem as solids_2_3 and solids_3_3 to related to this notion of dim and spacedim.

Reply:
Because DEMSolidObjects and RigidsSolidObjects are parameters handler object, they don't need the spacedim, and SerialObjects is actually the solid object....

Reply:
Damn... this is was I was so confused about all the names of those solid objects thing. The name of those class are not coherent haha